pkg/controller/certificates/rootcacertpublisher/publisher.go

```diff
@@ -171,8 +171,15 @@ func (c *Publisher) processNextWorkItem() bool {
 }
 
 func (c *Publisher) syncNamespace(ns string) error {
+	var err error
```

Comment:
this is getting shadowed by later declarations (if Create returns an error, it doesn't get stored in this var)

Reply:
done

Reply:
there are still paths that return a nil but leave err set to non-nil:

```
1. get returns NotFound (configmap doesn't exist)
2. create returns NotFound (namespace no longer exists)

This scenario returns nil, but leaves err set to non-nil and will record a 500
```

do we want a named return in this case so err is always set to the value being returned?

Reply:
oops, done